module/VuFind/src/VuFind/AjaxHandler/GetSideFacets.php

```diff
@@ -0,0 +1,263 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  AJAX
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\AjaxHandler;
+
+use VuFind\Recommend\PluginManager as RecommendPluginManager;
+use VuFind\Search\RecommendListener;
+use VuFind\Search\SearchRunner;
+use VuFind\Search\Solr\HierarchicalFacetHelper;
+use VuFind\Session\Settings as SessionSettings;
+use Zend\Mvc\Controller\Plugin\Params;
+use Zend\View\Renderer\RendererInterface;
+
+/**
+ * "Get Side Facets" AJAX handler
+ *
+ * @category VuFind
+ * @package  AJAX
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class GetSideFacets extends \VuFind\AjaxHandler\AbstractBase
+    implements \Zend\Log\LoggerAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+
+    /**
+     * Recommend plugin manager
+     *
+     * @var RecommendPluginManager
+     */
+    protected $recommendPluginManager;
+
+    /**
+     * Search runner
+     *
+     * @var SearchRunner
+     */
+    protected $searchRunner;
+
+    /**
+     * Hierarchical facet helper
+     *
+     * @var HierarchicalFacetHelper
+     */
+    protected $facetHelper;
+
+    /**
+     * Main facet configuration
+     *
+     * @var \VuFind\Config\PluginManager
+     */
+    protected $facetConfig;
+
+    /**
+     * View renderer
+     *
+     * @var RendererInterface
+     */
+    protected $renderer;
+
+    /**
+     * Constructor
+     *
+     * @param SessionSettings         $ss       Session settings
+     * @param RecommendPluginManager  $rpm      Recommend plugin manager
+     * @param SearchRunner            $sr       Search runner
+     * @param HierarchicalFacetHelper $fh       Facet helper
+     * @param \Zend\Config\Config     $fc       Facet config
+     * @param RendererInterface       $renderer View renderer
+     */
+    public function __construct(SessionSettings $ss,
+        \VuFind\Recommend\PluginManager $rpm,
+        SearchRunner $sr, HierarchicalFacetHelper $fh,
+        \Zend\Config\Config $fc, RendererInterface $renderer
+    ) {
+        $this->sessionSettings = $ss;
+        $this->recommendPluginManager = $rpm;
+        $this->searchRunner = $sr;
+        $this->facetHelper = $fh;
+        $this->facetConfig = $fc;
+        $this->renderer = $renderer;
+    }
+
+    /**
+     * Handle a request.
+     *
+     * @param Params $params Parameter helper from controller
+     *
+     * @return array [response data, HTTP status code]
+     */
+    public function handleRequest(Params $params)
+    {
```

Comment:
This is an extremely long method. Would it make sense to break it into smaller parts? Is any of this copied and pasted from elsewhere in a way that might be simplified through refactoring?

Reply:
(Also, let me know if you'd like me to take a stab at refactoring, if you're too busy -- I'm certainly willing, since it's a good way to review the code very closely).